### PR TITLE
feat(autoware_adapi_v1_msgs): new msg for fuel level

### PR DESCRIPTION
**Pull request by beyzanurkaya**
## Description

add msgs for fuel level 

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_adapi_v1_msgs/vehicle/msg/FuelLevel.msg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+builtin_interfaces/Time stamp
+float32 fuel_level # fuel level in percentage
```
Comment (xmfcx) [inline autoware_adapi_v1_msgs/vehicle/msg/FuelLevel.msg:2]:
```suggestion
float32 fuel_level # Fuel level in percentage of the total capacity (0.0 - empty, 100.0 - full)
```

While the current comments are sufficient, more detailed comments could provide additional clarity.

**Comment by xmfcx** [comment]
It might be a good idea to have a more generic message to cover any fuel source. And support for hybrid cars.

```
# EnergyLevel.msg

builtin_interfaces/Time stamp # Time stamp of the message creation or event occurrence

# Constants representing energy source types
uint8 ENERGY_SOURCE_BATTERY = 0
uint8 ENERGY_SOURCE_FUEL = 1
uint8 ENERGY_SOURCE_HYDROGEN = 2

# Energy source type
uint8 energy_source 

# Current energy level in percentage of the total capacity (0.0 - empty, 100.0 - full)
float32 energy_level_percentage 

# Actual energy level, expressed in the appropriate units for the energy source (e.g., liters for fuel, kWh for battery, kg for hydrogen)
float32 energy_level_actual 

# Total capacity of the system, expressed in the appropriate units for the energy source (e.g., liters for fuel, kWh for battery, kg for hydrogen)
float32 energy_capacity 
```

And a new message for hybrid  vehicles:

```
# HybridVehicleEnergyLevel.msg

# Array of EnergyLevel messages, one for each energy source in the vehicle
EnergyLevel[] energy_levels
```

**Comment by isamu-takagi** [comment]
Energy level is already defined in [VehicleStatus.msg](https://github.com/autowarefoundation/autoware_adapi_msgs/blob/main/autoware_adapi_v1_msgs/vehicle/msg/VehicleStatus.msg#L6) and should be updated with the new message if needed. Also, it might be a good to separate dynamic data (energy_level_percentage, energy_level_actual) as topic and static data (energy_source, energy_capacity) as service or transient_local topic.

**Comment by mehmetdogru** [comment]
Thank you @isamu-takagi , as you pointed out I guess already existing one should be enough for our case and if needed an extension to the message can be provided with another PR later on.

**Pull request closed without merging**